api/src/libs/githubInvitationInit.ts

```diff
@@ -0,0 +1,162 @@
+import { createAppAuth } from '@octokit/auth'
+import { request } from '@octokit/request'
+import { getGithubPrivateKey } from '../config/githubConfig'
+import { logger } from '@bcgov/common-nodejs-utils';
+
+// cached value
+type APPInitailValue = {
+  initialized: number | null,
+  apps: object,
+  nonInstallatedApp: Function | null,
+}
+
+const  installationApps = <APPInitailValue>{
+  initialized: null,
+  apps: {},
+  nonInstallatedApp: null
+}
+
+/**
+ * getNonInstallationApp
+ * @returns a non installed github app
+ */
+export const getNonInstallationApp = () => {
+    logger.info('getNonInstallationApp')
+  // caches a non installed app
+  try{
+  if (!installationApps.nonInstallatedApp) {
+    const auth = createAppAuth({
+      appId: process.env.GITHUB_APP_ID ||  131752,
+      privateKey: getGithubPrivateKey(),
+      clientId: process.env.GITHUB_CLIENT_ID,
+      clientSecret: process.env.GITHUB_CLIENT_SECRET,
+    })
+
+    installationApps.nonInstallatedApp = request.defaults({
+      request: {
+        hook: auth.hook,
+      },
+      mediaType: {
+        previews: ['machine-man'],
+      },
+    })
+  }
+ } catch(e){
+    logger.error('I catch an error', e);
```

Comment:
Grammar, but in reality a more useful log message would be helpful: 
```suggestion
    logger.error('I caught an error', e);
```